### ISSUE TITLE
[BUG] Version 4 does not seem to respect default lang

### ISSUE THREAD

**Issue by sbonardt** [AnimalDesign/bolt-translate#87]
I've updated my (working) Translate extension from 3.1 on a Bolt 3 to the latest version 4 (with latest Bolt) but it doesn't seem to recognise the default language set in the Bolt main config file anymore.

I've set the Bolt main locale in config to: nl_NL

In the extension config (translate.animal.yml) I've set:

```
# The locales the site should be available in
locales:
    nl_Nl:
        label: NL
        slug: nl
    en_GB:
        label: EN
        slug: en
```

And thus made sure the default is the first one in order. (nl_NL)

And in contenttypes.yml i've set the required locale fields in the contenttypes:

```
        locale:
            type: locale
            group: main
        nldata:
            type: hidden
        nlslug:
            type: locale_data
            index: true
        endata:
            type: hidden        
        enslug:
            type: locale_data
            index: true
```

When I load the homepage the default go to language is however en_GB and not nl_NL as expected. Also in the backend '?_locale=en' keeps getting added as default lang.
I expect to see the default language nl_NL loaded in frontend as well as backend.

It looks like I've set all required settings and config. (?)

Tested with Bolt 3.1.5 and Translate versions 4.0.1 and 4.0.0. Going back to versions 3.x.x doesn't work with Bolt 3.1.5

**Comment by sbonardt** [comment]
Retested on a fresh install and it seemed to give the same results. However, on rechecking and rereading (multiple times...) I noticed the 'use_accept_language_header' setting in the config.

By default, this is set to TRUE. 

```
use_accept_language_header: false
```

I would advise to set this to default state 'false', and make the 'use_accept_language_header' something you choose for as your setting up, OR change the README documentation stating that the first locale is the one that is the default one. This is not the case when this setting is set to 'true'.

Background: I'm a dutchie working with a browser(s) with a never changed locale setting (so not explicitly set to 'en'), but wanting the website to show NL lang by default. Had me in bits, because reading the documentation did not clarify it enough for me.

**Comment by sbonardt** [comment]
**TL;DR:** 
**For version 4 and up...**
Is your website not showing your default language you've set in main config.yml file and in translate extension config (by setting preferred lag first). Check the 'use_accept_language_header' setting in the extensions config file. If set to TRUE this overrides everything.

**Comment by SahAssar** [comment]
@sbonardt I get that this is causing grief... Unfortunately I think this is one of those no-win scenarios. How to "detect" preferred language is one of those things on the internet that everyone seems to do poorly and that there is no good solution to. We could make it default to off though (as you suggested), @peterverraedt are you okay with that?

**Comment by peterverraedt** [comment]
@SahAssar I'm okay with that, defaulting to off would definitely avoid this kind of problems. It is probably best if we add some documentation on this feature in the Readme though, so that future users know this feature exists. 

I actually think that this "misbehaviour" of detecting the preferred language only exists in Dutch speaking countries and Scandinavian countries. Other countries tend to set their operating system language to their mother tongue :).
